doc/source/dev_guide/plugins.rst

```diff
@@ -1,34 +1,207 @@
-================================================
- Adding new functionality to Satpy via plugins
-================================================
+===========================
+Extending Satpy via plugins
+===========================
 
 .. warning::
     This feature is experimental and being modified without warnings.
     For now, it should not be used for anything else than toy examples and
     should not be relied on.
 
-Satpy has the capability of using plugins. At the moment, new composites can be
-added to satpy through external plugins. Plugins for reader and writers may be
-added at a later date (PRs are welcome!).
+Satpy is able to load additional functionality outside of the builtin features
+in the library. It does this by searching a series of configured paths for
+additional configuration files for:
 
-Here is an
-`example <https://github.com/mraspaud/satpy-composites-plugin-example>`_ of a
-composites plugin.
+* readers
+* composites and modifiers
+* enhancements
+* writers
 
-The key is to use the same configuration directory structure as satpy and add
-a `satpy.composites` entry point in the setup.py file of the plugin:
+For basic testing and temporary configuration changes, you can follow
+the instructions in :ref:`component_configuration`. This will tell Satpy
+where to look for your custom YAML configuration files and import any Python
+code you'd like it to use for these components. However, this requires telling
+Satpy of these paths on every execution (either as an environment variable or
+by using ``satpy.config``).
 
-.. code: python
+Satpy also supports being told this information via setuptools "entry points".
+Once your custom Python package with entry points is installed Satpy will
+automatically discover it when searching for composites without the user
+needing to explicitly import your package. This has the added
+benefit of organizing your YAML configuration files and any custom python code
+into a single python package. How to structure a package in this way is
+described below.
+
+An example project showing the usage of these entry points is available at
+`this github repository <https://github.com/pytroll/satpy-composites-plugin-example>`_
+where a custom compositor is created. This repository also includes common
+configuration files and tools for writing clean code and automatically testing
+your python code.
```

Comment:
Thats very nice!!!

Reply:
I realize now that I had intended including a pre-commit config, but decided against it in the last minute to avoid too much confusion.